front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/handleParseFiles.ts

```diff
@@ -9,71 +9,94 @@ export const handleFileReadingError = (error: Error) => {
   console.error('File reading error:', error);
 };
 
-export const handleJsonParsingError = (error: Error, dispatch: Dispatch) => {
-  console.error('Error parsing JSON:', error);
-  dispatch(
-    setFailure({
-      name: t('errorMessages.error'),
-      message: t('errorMessages.errorInvalidJSONFormat'),
-    })
-  );
-};
+const validateTrainSchedules = (
+  importedTrainSchedules: Record<string, unknown>[]
```

Comment:
This should work to avoid many `any` in the function ?
```suggestion
  importedTrainSchedules: Partial<TrainScheduleBase>[]
```

Reply:
Done :)